gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/util/UserAgent.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.driver.util;
+
+import org.apache.tinkerpop.gremlin.util.Gremlin;
+
+import javax.naming.NamingException;
+
+public class UserAgent {
+    public static String getUserAgent() {
+        String applicationName = null;
+        try {
+            applicationName = (String)(new javax.naming.InitialContext().lookup("java:app/AppName"));
+        } catch (NamingException e) {
+            applicationName = "NotAvailable";
+        };
+        String glvVersion = Gremlin.version().replace(' ', '_');
+        String javaVersion = System.getProperty("java.version").replace(' ', '_');
+        String osName = System.getProperty("os.name").replace(' ', '_');
+        String osVersion = System.getProperty("os.version").replace(' ', '_');
+        String cpuArch = System.getProperty("os.arch").replace(' ', '_');
```

Comment:
I believe that all variables should be declared final if they aren't going to be modified later. 
```suggestion
        final String glvVersion = Gremlin.version().replace(' ', '_');
        final String javaVersion = System.getProperty("java.version").replace(' ', '_');
        final String osName = System.getProperty("os.name").replace(' ', '_');
        final String osVersion = System.getProperty("os.version").replace(' ', '_');
        final String cpuArch = System.getProperty("os.arch").replace(' ', '_');
```
Also, would there be a case that these gets return empty string or null?

Reply:
I can make them all final.
These properties should always be defined but it seems I can provide a default value of "NotAvailable". I will make that change